pdpserver/server/logger.go

```diff
@@ -0,0 +1,40 @@
+package server
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/infobloxopen/themis/pdp"
+)
+
+const lowestVerbosityLevel = logrus.PanicLevel
+
+// LevelCheckable interface ensures public getter for log level
+type LevelCheckable interface {
+	GetLevel() logrus.Level
+}
+
+// DetailedInfo interface provides detailed information consumed during WithFields.
+// 	- GetDetail should be an expensive operation that provides more information
+// 	- FilterLevel should be checked against the logger level to ensure that details
+//	  are only processed when necessary
+//	- String ensures DetailedInfo provides a brief description in the event
+//	  that the logger does not consume through GetDetail
+type DetailedInfo interface {
+	GetDetail(nShow uint) string
+	FilterLevel() logrus.Level
+	String() string
+}
+
+// compile-time interface check
+var (
+	_ DetailedInfo = pdp.PolicyUpdateDetail{}
```

Comment:
The line gives a clue that DetailedInfo interface belongs to pdp not to here. But is it needed at all? I see it as a way to extend number of log levels to different dimension. Why does it appear that levels which logrus provide out of the box aren't enough? I still prefer to not solve the issue by hacks as this. But instead introduce proper set debug tools. One of them should be able to get policy by id from pdp.